ibis/expr/datatypes/core.py

```diff
@@ -1692,3 +1726,19 @@ def _geom_to_wkt(
         base_geom: geom.base.BaseGeometry,
     ) -> _WellKnownText:
         return _WellKnownText(base_geom.wkt)
+
+
+class NonNullables(Singleton):
+    """Namespace class for convenient construction of non-nullable types."""
+
+    def __getattr__(self, name: str) -> functools.partial:
+        import ibis.expr.datatypes as dt
+
+        ty = getattr(dt, name)
+        if not issubclass(ty, dt.DataType):
+            raise AttributeError(name)
+
+        return functools.partial(ty, nullable=False)
+
+
+public(n=NonNullables())
```

Comment:
I may miss something but I rather dislike this syntax sugar due to the following reasons:
- why is `dt.n.Int64()` is better then `dt.Int64(nullable=False)`?
- why to expose it publicly?
- confusing shorthand, I'd rather think that `dt.n` means `nullable` and `dt.nn` means `not nullable`

Reply:
The PR is a nice improvement otherwise!

Reply:
Yeah, this was just an experiment in making it more convenient. I'll revert the `n` syntax and keep it verbose for now.

Reply:
Done.